integration/containerd/confidential/agent_image.bats

```diff
@@ -94,6 +94,18 @@ setup() {
 	echo "Reconfigure Kata Containers"
 	switch_image_service_offload on
 	clear_kernel_params
+
+        local_dns=$(grep nameserver /etc/resolv.conf \
+            /run/systemd/resolve/resolv.conf  2>/dev/null \
+            |grep -v "127.0.0.53" | cut -d " " -f 2 | head -n 1)
+
+        if [ ! -z "$HTTPS_PROXY" ]; then
+	    add_kernel_params "agent.https_proxy=$HTTPS_PROXY"
+            sed -i -e 's/8.8.8.8/'${local_dns}'/' "${pod_config}"
```

Comment:
Let me take that opportunity to ask some questions about DNS for CC use case.

When I created those confidential tests I realized the built guest image was configured to use the host's DNS address already, but name resolution was failing.  I lack knowledge on that area, but at that time I assumed I was not configuring the CNI plug-in properly. So, first question, in order to make the dns working with CRI + Containerd (no k8s on the equation) is it only a matter of configure the cni plug-in correctly? If yes, how?

The "solution" that I took was to hard-code to `8.8.8.8` on the pod configuration file. This seems wrong in real world though, users shouldn't need to set a dns on their pod configs. So how to proper implement that?

Reply:
@wainersm Yes, guest image was configured to use the host's DNS config: `/etc/resolv.conf` but when host OS enable `systemd-resolved` service, the real DNS config will under `/run/systemd/resolve/resolv.conf` and in `/etc/resolv.conf` will be: `nameserver 127.0.0.53`, and in kata guest, there will be no `systemd-resolved` service and related config which make DNS failed to work even the network connection is ok, hard-code `8.8.8.8` on the pod configuration file can fix this issue.

We can test at kubernetes level which will install the coredns services to resolve the dns issue for pod.
We can also fix the image generation script to use the real dns config for different OS.